sema4ai/src/sema4ai_code/robo/collect_actions.py

```diff
@@ -135,39 +191,65 @@ def collect_actions_full_and_slow(
     else:
         cwd = str(path)
 
-    error_msg = "Unable to collect @actions"
-    try:
-        for ep in pm.get_implementations(EPResolveInterpreter):
-            interpreter_info: IInterpreterInfo = ep.get_interpreter_info_for_doc_uri(
-                uri
-            )
-            if interpreter_info is not None:
-                environ = interpreter_info.get_environ()
-                python_exe = interpreter_info.get_python_exe()
-                future = launch_and_return_future(
-                    [python_exe] + args,
-                    environ=environ,
-                    cwd=cwd,
-                    timeout=30,
-                    monitor=monitor,
-                )
-                result = future.result(30)
-                if result.returncode == 0:
-                    if result.stdout:
-                        try:
-                            return ActionResult.make_success(json.loads(result.stdout))
-                        except Exception:
-                            msg = f"Unable to parse as json: {result.stdout}"
-                            log.exception(msg)
-                            return ActionResult.make_failure(msg)
-                if result.stderr:
-                    error_msg = (
-                        f"Found errors while collecting actions errors: {result.stderr}"
-                    )
-                    log.info(error_msg)
-                break
-    except BaseException as e:
-        log.exception("Error collection @action")
-        return ActionResult.make_failure(f"Unable to collect @actions: {e}")
+    return _execute_within_user_env(pm, uri, args, monitor, cwd)
 
-    return ActionResult.make_failure(error_msg)
+
+def _get_actions_version(
+    pm: PluginManager, uri: str, monitor: IMonitor
+) -> ActionResult:
+    from pathlib import Path
+
+    from sema4ai_ls_core import uris
+
+    libname = "sema4ai.actions"
+    args = ["-c", f"import {libname};print({libname}.__version__)"]
+
+    path = Path(uris.to_fs_path(uri))
+    if not path.is_dir():
+        cwd = str(path.parent)
+    else:
+        cwd = str(path)
+
+    error_msg = f"""Unable to get {libname} version.
+
+    This usually means that `{libname}` is not installed in the python
+    environment (make sure that `{libname.replace('.', '-')}`
+    is defined in your `package.yaml`).
+    """
+
+    result = _execute_within_user_env(pm, uri, args, monitor, cwd, returns_json=False)
+    if result.success and result.result:
+        try:
+            result.result = tuple(int(x) for x in result.result.strip().split("."))
+        except Exception:
+            result.message = error_msg
```

Comment:
I think you want to return a new `ActionResult` here.

Otherwise you also have to set `result.success` to `False`